chingu-x/chingu-dashboard-be
# Feature/oauth GitHub

# Description

### To test this PR
1. obtain github oauth client id and client secret (on clickup, you can also make your own for dev), and put them in .env
2. `npx prisma db push` for both dev and test db
3. go to `http://localhost:8000/api/v1/auth/github/login` a github authorization table will pop up, and it should return the user's basic github profile 


## Issue link

Fixes # (issue)

## Type of change



# How Has This Been Tested?

Manually and automated tests

# Checklist:

- [ x] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my code
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ x] I have made corresponding changes to the documentation
- [ x] My changes generate no new warnings
- [ x] I have added tests that prove my fix is effective or that my feature works
- [ x] New and existing tests pass locally with my changes
- [ x] Any dependent changes have been merged and published in downstream modules
- [ x] I have updated the change log
